src/main/java/esme/Ui.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    /**
+     * Checks if the given task index is valid in terms of the task list.
+     * 
+     * @param index The index to check.
+     * @return True if the index is valid, false otherwise.
+     */
+    public boolean isIndexValid(int index) {
+        return index <= taskList.getNumberOfTasks() && index >= 0;
+    }
+
+    /**
+     * Marks the task at the given index as done. The index is 1-indexed since
+     * arrays are 0-indexed.
+     * 
+     * @param taskIndex The index of the task to be marked as done.
+     */
+    public void markTaskInList(int taskIndex) {
+        taskList.markTask(taskIndex - 1);
+        displayLine(true);
+        System.out.println("\tOutstanding! This task is marked as done, and your destiny shines brighter.");
+        System.out.println("\t  [X] " + taskList.getTask(taskIndex - 1).getDescription());
+        displayLine(true);
+    }
+
+    /**
+     * Unmarks the task at the given index as done. The index is 1-indexed since
+     * arrays are 0-indexed.
+     * 
+     * @param taskIndex The index of the task to be marked as undone.
+     */
+    public void unmarkTaskInList(int taskIndex) {
+        taskList.unmarkTask(taskIndex - 1);
+        displayLine(true);
+        System.out.println("\tFear not, for this task remains unfinished. We shall conquer it in due time!");
+        System.out.println("\t  [] " + taskList.getTask(taskIndex - 1).getDescription());
+        displayLine(true);
+    }
+
+    /**
+     * Handles the task status command given in the words array.
+     * If the command is "mark", it marks the task at the given index.
+     * If the command is "unmark", it unmarks the task at the given index.
+     * The index should be a valid index in the list, and the command should
+     * be one of the above two.
+     * If the index is out of range or the command is invalid, it prints an
+     * appropriate error message.
+     * 
+     * @param words The words array containing the command and the index.
+     */
+    public void handleTaskStatus(String[] words) {
+        String command = words[0];
+        try {
+            int index = Integer.parseInt(words[1]);
+            if (isIndexValid(index)) {
+                toggleTaskStatus(index,command);
+            } else {
+                System.out.println("Oh dear, it seems the index has wandered beyond the boundaries of our list!");
+            }
+        } catch (Exception e) {
+            System.out.println("Error: Wrong format! Please use the format: command index (e.g., '" + command + " 1')");
+        }
+    }
+
+    public void toggleTaskStatus(int taskIndex, String command) {
+        if (command.equals("mark")) {
+            markTaskInList(taskIndex);
+        } else {
+            unmarkTaskInList(taskIndex);
+        }
+    }
+
+    public void printTaskList() {
+        taskList.printTaskList();
+        displayLine(true);
+    }
+
+    public String esmeLogo = " _____                    \n" +
```

Comment:
Since the logo won't change in the whole process, is it more appropriate to define it as a constant? For example ``private static final String ESME_LOGO``